### PR TITLE
Allow Overriding of ToC

### PR DESCRIPTION
**Pull request by JannikArndt**
Motivation: I'd like to use the title of a post as the title of the ToC as well (since I have it floating left of the text).

Problem: The "outer" part of the ToC (which includes the title) is currently defined in the `single.html`, and only the `<ul>` is defined in the `toc.html`-partial.

This PR moves the complete html-code of the ToC into the `toc.html` partial.

* note: The condition _if_ the ToC is displayed stays inside the `single.html`, to be consistent with with other conditional partials `comments` and `share_icons`.
* note: I did _not_ re-indent the `toc.html` to keep the diff easier. If you prefer proper indentation, I'll gladly do that 👍

### CODE REVIEW

**Comment by kdkasad** [comment]
> This PR moves the complete html-code of the ToC into the `toc.html` partial.

I like this idea.

> * note: I did _not_ re-indent the `toc.html` to keep the diff easier. If you prefer proper indentation, I'll gladly do that +1

I'd say add the proper indentation but keep it as a separate commit. That way it's still easy to see what's changed by just not looking at that commit (or by using `--word-diff`).

File: assets/css/common/post-single.css [hunk 1]
```diff
@@ -210,7 +210,7 @@
     display: block;
     margin: auto 0;
     padding: 10px;
-    color: rgba(255, 255, 255, .8);
+    color: var(--primary);
```
Comment (kdkasad) [inline assets/css/common/post-single.css:213]: Why is this changed? Does it have anything to do with the rest of the PR?

**Edit:** it appears to break the look of code blocks in light mode.

**Review by kdkasad — CHANGES_REQUESTED** [review]
Everything looks good except for that one style change. Testing all looks fine; nothing seems to change.

File: assets/css/common/post-single.css (hunk 1 above)
Comment (JannikArndt) [reply assets/css/common/post-single.css]: Arg, I should not have used my `master` branch for the PR 🙈 I reverted this change there 👍 (b46ada5)

**Comment by JannikArndt** [comment]
>I'd say add the proper indentation but keep it as a separate commit.

Done in 8d3c7fd 👍

File: assets/css/common/post-single.css (hunk 1 above)
Comment (kdkasad) [reply assets/css/common/post-single.css]: Thanks! It might make sense to rebase on top of upstream's master, so there aren't two extra commits, but that's up to you and/or @adityatelange.

**Comment by kdkasad** [comment]
> > I'd say add the proper indentation but keep it as a separate commit.
> 
> Done in [8d3c7fd](https://github.com/adityatelange/hugo-PaperMod/commit/8d3c7fddf5103f5b0e28853cb8cbc6633ee62b4a)

Awesome, thanks! I ran `git diff -w` and it shows that only whitespace has changed, so even though it's a large commit, it's all good.

**Review by kdkasad — APPROVED** [review]
Everything looks good now. No side effects appeared during testing.

I say it's ready to merge.

**Comment by JannikArndt** [comment]
Haha, I usually just click the green "Squash and Merge" for this, but okay, I gathered all my git knowledge and did a rebase 🤪 

If you ask GitHub to hide whitespace changes, it shows the actual changes pretty well:
image

**Review by adityatelange — APPROVED** [review]
